src/main/java/htsjdk/samtools/util/BlockCompressedInputStream.java

```diff
@@ -587,41 +586,98 @@ private int unpackInt32(final byte[] buffer, final int offset) {
 
     public enum FileTermination {HAS_TERMINATOR_BLOCK, HAS_HEALTHY_LAST_BLOCK, DEFECTIVE}
 
+    /**
+     *
+     * @param file the file to check
+     * @return status of the last compressed block
+     * @throws IOException
+     */
     public static FileTermination checkTermination(final File file) throws IOException {
-        final long fileSize = file.length();
+        return checkTermination(file == null ? null : file.toPath());
```

Comment:
@pshapiro4broad It absolutely just shifts the npr to later.  I wanted to just keep the exact behavior of the original function, which was to NPR somewhere later.  It does probably makes sense to check for null and throw a nice exception instead of an NPR.  Returning defective would be a bad idea for a null file I think since it would hide other errors.